app/services/post_alerter.rb

```diff
@@ -640,7 +640,15 @@ def notify_pm_users(post, reply_to_user, notified)
 
   def group_notifying_via_smtp(post)
     return nil if !SiteSetting.enable_smtp || post.post_type != Post.types[:regular]
-    post.topic.allowed_groups.where(smtp_enabled: true).first
+    if post.topic.allowed_groups.count == 1
+      return post.topic.first_smtp_enabled_group
+    end
+
+    group = Group.find_by_email(post.topic.incoming_email.first.to_addresses)
```

Comment:
@martin-brennan Looking at the error logs: `post.topic.incoming_email.first` can be nil

Reply:
https://github.com/discourse/discourse/pull/14981